### ISSUE TITLE
val_acc does not change during LSTM time series binary classification training 

### ISSUE THREAD

**Issue by DSA101** [fchollet/keras#2941]
While doing training of LSTM time series binary classification model, val_acc does not change during training, while loss, acc and val_loss do change. Later I've found that this issue is usually the case when there are too few (e.g. 330) sequences are provided for training. But still how val_acc can stay exactly the same despite training and validating on different batches?

model = Sequential()
model.add(LSTM(256, input_shape=(6, 10)))
model.add(Dropout(0.5))
model.add(Dense(1))
model.add(Activation('sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='rmsprop', metrics=['accuracy'])
hist = model.fit(X_train_mat, Y_train_mat, nb_epoch=10000, batch_size=30, validation_split=0.1)

Epoch 2816/10000
 50/472 [==>...........................] - ETA: 0s - loss: 0.6281 - acc: 0.6800Epoch 02815: val_acc did not improve
472/472 [==============================] - 0s - loss: 0.5151 - acc: 0.7648 - val_loss: 1.2978 - val_acc: 0.4151
Epoch 2817/10000
 50/472 [==>...........................] - ETA: 0s - loss: 0.4406 - acc: 0.8600Epoch 02816: val_acc did not improve
472/472 [==============================] - 0s - loss: 0.5179 - acc: 0.7479 - val_loss: 1.2844 - val_acc: 0.4151
Epoch 2818/10000
 50/472 [==>...........................] - ETA: 0s - loss: 0.5385 - acc: 0.7400Epoch 02817: val_acc did not improve
472/472 [==============================] - 0s - loss: 0.5100 - acc: 0.7585 - val_loss: 1.2699 - val_acc: 0.4151

**Comment by ersinyar** [comment]
For binary classification the val_acc values you obtain are worse than random. So, this may be related to indexing the right output values for corresponding input samples.
